tests/test_cnn.py

```diff
@@ -35,3 +38,113 @@ def test_cnn(tmp_path, model_class):
     assert np.allclose(action, model.predict(obs, deterministic=True)[0])
 
     os.remove(str(tmp_path / SAVE_NAME))
+
+
+def patch_dqn_names_(model):
+    # Small hack to make the test work with DQN
+    if isinstance(model, DQN):
+        model.critic = model.q_net
+        model.critic_target = model.q_net_target
+
+
+def params_should_match(params, other_params):
+    for param, other_param in zip_strict(params, other_params):
+        assert th.allclose(param, other_param)
+
+
+def params_should_differ(params, other_params):
+    for param, other_param in zip_strict(params, other_params):
+        assert not th.allclose(param, other_param)
+
+
+def check_td3_feature_extractor_match(model):
+    for (key, actor_param), critic_param in zip(model.actor_target.named_parameters(), model.critic_target.parameters()):
+        if "features_extractor" in key:
+            assert th.allclose(actor_param, critic_param)
+
+
+@pytest.mark.parametrize("model_class", [SAC, TD3, DQN])
+def test_feature_extractor_target_net(model_class):
```

Comment:
I found this function hard to follow. That's not a big deal, it's test code. I think suggestion (below) to take list of attribute-pairs on a per-algorithm case would reduce special cases making it easier to follow. But if it's still convoluted, I wonder if it's actually better to just bite some code duplication and have a different test for each class? (We might be able to factor out some common code.)

Reply:
I do agree this is quite hard to follow (I wrote it on the fly without the big picture in mind ^^")
On the other hand, the checks are tricky anyway... I will see what I can do, maybe one test per algorithm yes.